packages/aws-cdk-lib/aws-autoscaling/lib/step-scaling-policy.ts

```diff
@@ -102,6 +119,18 @@ export class StepScalingPolicy extends Construct {
       throw new Error(`'scalingSteps' can have at most 40 steps, got ${props.scalingSteps.length}`);
     }
 
+    if (props.datapointsToAlarm !== undefined && !Token.isUnresolved(props.datapointsToAlarm)) {
+      if (props.datapointsToAlarm < 1) {
+        throw new Error(`datapointsToAlarm cannot be less than 1, got: ${props.datapointsToAlarm}`);
+      }
+      if (props.evaluationPeriods !== undefined
+        && !Token.isUnresolved(props.evaluationPeriods)
+        && props.evaluationPeriods < props.datapointsToAlarm
```

Comment:
Should we throw an error if `datapointsToAlarm` is set without setting `evaluationPeriods`. I know `evaluationPeriods` can be set by itself and changing that behavior would break compatibility but we should probably add that check for `datapointsToAlarm`. Can't find a use case for setting `datapointsToAlarm` by itself.

Reply:
I agree. I will add the validation for only `datapointsToAlarm`.